packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -718,12 +719,33 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
 
   bool _warmUpFrame = false;
 
+  void _warmUpSkiaShaderCompilations({ShaderWarmUp customShaderWarmUp}) {
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+
+    if (customShaderWarmUp != null) {
+      customShaderWarmUp(canvas);
+    } else {
+      defaultShaderWarmUp(canvas);
+    }
+
+    final Picture picture = recorder.endRecording();
+    final TimelineTask shaderWarmUpTask = TimelineTask();
+    shaderWarmUpTask.start('Warm-up shader');
+    picture.toImage(1000, 1000).then((Image image){
+      shaderWarmUpTask.finish();
```

Comment:
this doesn't finish the shaderWarmUpTask, it finishes the last task that was started. Since we don't guarantee ordering here I think that means that we're not going to get coherent results in practice.

Reply:
nevermind, i misread the docs.